### ISSUE TITLE
Добавить функциональность быстрого поиска из браузера

### ISSUE THREAD

**Issue by akmetainfo** [OpenCorpora/opencorpora#139]
Часто при выполнении заданий хочется посмотреть-проверить, есть ли слово в словаре OpenCorpora, а если есть -- то посмотреть их все.

Мне кажется, было бы удобно добавить такую фичу в проект. Выделил слово -- правый клик -- и сразу в новой вкладке открываются результаты поиска.

В принципе, я уже себе сделал, пользуюсь. (См. https://github.com/akmetainfo/opencorpora/issues/3 ) -- но это было так, вчерновую сделано. Насколько я понял -- сейчас быстрый поиск в браузерах на OpeSearch делают.

**Comment by madfriend** [comment]
cc @grandsbor:
нужно?

**Comment by grandsbor** [comment]
Честно сказать, не знаю. Если придумать одновременно понятный и не мешающий разметке интерфейс этого поиска, я не против.

**Comment by akmetainfo** [comment]
@grandsbor Дмитрий, по вашим словам "придумать интерфейс" мне кажется вы не понимаете о чём речь. Это уже вшито в современные браузеры, не нужно делать свой интерфейс.

Мгм. Лучше один раз увидеть, чем сто раз услышать. Вот скриншот как это сделано сейчас:
![123](https://cloud.githubusercontent.com/assets/6085279/5815953/d77801ea-a0ab-11e4-96f6-0b6f59679ecd.png)

**Comment by grandsbor** [comment]
Ну ждём тогда пулл-реквест

**Comment by madfriend** [comment]
А есть уверенность, что фичу можно прикрутить ко всем браузерам и более-менее единым образом?

**Comment by akmetainfo** [comment]
Я уже лет пять-семь пишу 'под себя' такие поисковые плагины под Firefox и надо сказать даже не задумывался, что с другими браузерами. Беглое гугление показывает, что я написал плагин в формате 'Sherlock' и этот формат объявлен уже год как deprecated (но сейчас у меня всё работает в последней версии Firefox).

Википедия [подсказывает](https://ru.wikipedia.org/wiki/OpenSearch), что новый поисковый формат (OpenSearch) поддерживается и другими браузерами (FF, IE7+, Chrome). На нём я не писал, но вроде ничего сложного.

Перепилю на OpenSearch -- и можно считать, что к большинству браузеров (не всем) прикручено единообразно.

Меня больше интересует востребованность фичи, а не техническая сторона.
Потенциально фича может быть востребована в двух кейсах:
1. Опытный размечающий проверяет, какие пометы стоят в словаре у слова. (Почему не в тот пул попало и т.п.) Надеюсь, понятно, что имеется ввиду.
2. Мы даём возможность всему интернету искать по словарю опенкорпоры. Подобных словарей в интернете достаточно, можно в дальнейшем предлагать не только данные из нашей базы, но и ссылку допустим на викисловарь.

**Comment by madfriend** [comment]
И еще у меня вопрос. Получается, это плагин к сайту, и он никак не завязан на внутренний код? Если так, то может и вынести его в отдельный репозиторий?

**Comment by akmetainfo** [comment]
В текущей реализации (Sherlock) завязка на код сайта слабая: ну есть некоторая статическая страница с текстом (ничего не мешает её положить на другой сайт), есть текстовый файл описания плагина. В этом описании по-факту сказано "вот по этому адресу есть страница, если на неё сделать POST-запрос, с параметром вот таким -- то получишь результат поиска".

Но привязка всё же есть. Заходит пользователь на сайт (наш, википедии) а ему браузер предлагает - о, этот сайт имеет несколько поисковых машин, хочешь добавить себе??

Если же речь про OpenSearch, то, насколько я понял когда пролистал документацию -- нужно будет дополнительно создать страницу поиска, чтобы она выдавала результаты поиска в строго определённой XML-ке. (Поэтому нужно будет сделать отдельный поиск из браузера, либо думать, как объединить с текущим поиском.) Связанность с кодом будет выше.

**Comment by akmetainfo** [comment]
> Upd: а из репозитория уже публиковать в библиотеки расширений для браузеров

В том-то и дело, что это не плагин и нет таких страниц на сайте Mozill'ы, откуда можно было бы поставить такие поисковые движки (раньше, лет пять назад -- да, были, добровольцы свои предлагали (некоторые даже включали), там был очень широкий каталог, я себе набрал кучу готовых поисковичков. Потом эту полезную штуку прикрыли, к сожалению)

Текущий вариант: ты зашёл на сайт -- сайт говорит "вот, у меня есть две поисковых машины, хочешь себе какую-нибудь?"

Поэтому можно и в отдельный репозиторий выделить (хотя не вижу смысла), но на сайт корпоры нужно будет залить по-любому.

**Comment by madfriend** [comment]
Раз прикрыли эту фичу, значит, возникает вопрос о востребованности ее и для нашей аудитории.
Информацию о расширении, если оно будет, можно будет поместить в FAQ или на NLPub.

**Comment by akmetainfo** [comment]
В общем, ясно ))) Никто не пользуется такой штукой и не понимает о чём речь и как с ней работать. Дмитрий не понимает, вы тоже не понимаете.

Проще забить и подождать, пока кому-нибудь ещё понадобится.

По существу вопроса: фичу не прикрывали. Sherlock -> OpenSearch это не закрытие фичи, а переход на новую фичу (старая прекрасно поддерживается). Каталог же, ой... сейчас [вот во что преобразовался](https://addons.mozilla.org/ru/firefox/search/?atype=4), всё правильно вы говорили -- есть библиотеки для браузеров.

Эх, мама, у меня даже логин остался на addons.mozilla и расширение которое я делал до сих пор кто-то качает... Фигасе, сколько лет прошло...

**Comment by grandsbor** [comment]
Нет, ну когда у меня была Мозилла, я на многих сайтах пользовался такой штукой, удобно. Почему бы не сделать, особенно, если это несложно.

**Comment by akmetainfo** [comment]
Мне надо будет подумать ещё над данной фичей. Всем спасибо за конструктивную критику и замечания. Пока топик не закрывайте, я позже сюда ещё напишу.

**Comment by akmetainfo** [comment]
В общем, сделал расширение для Firefox, которое добавляет поиск по словарю корпоры в список поисковых машин браузера. (Ну, "расширение" - это громко сказано: по сути -- одна-единственная xml-ка. Даже не стал городить огород и делать отдельный репозиторий для этого.)
[Само расширение находится тут](https://addons.mozilla.org/ru/firefox/addon/opencorporadictionarysearch/). Для других браузеров так пока ничего и не решил.
Таску эту закрываю. Если появится желание продолжить развитие -- откроем заново.
Или пишите тут: https://github.com/akmetainfo/opencorpora/issues/3

**Comment by grandsbor** [comment]
Давайте, что ли, дадим на него ссылку со страницы поиска по словарю

**Comment by akmetainfo** [comment]
Ну, если доля FF'шных пользователей в статистике заметна -- то можно и на страницу словаря закинуть. А какие вообще браузеры используют размечающие? Ну, чтобы на будущее посматривать для других браузеров технологию.

**Comment by grandsbor** [comment]
Конкретно размечающих просто так не посмотреть. По всем посетителям за квартал:
- 40% Хром
- 28% FF
Остальные меньше 10%

**Comment by grandsbor** [comment]
Я оставил этот тикет, чтобы добавить ссылку :)

**Comment by akmetainfo** [comment]
Запросил полную проверку расширения у mozilla

**Comment by grandsbor** [comment]
И как оно?

**Comment by akmetainfo** [comment]
Полная проверка пройдена успешно, теперь при установке в браузер не выкидывается никаких предупреждений "расширение не проверено, вы устанавливаете на свой страх и риск".
